RecoTracker/LSTCore/src/alpaka/Segment.h

```diff
@@ -12,168 +14,65 @@
-          dPhiChangeMins_buf(allocBufWrapper<FPX>(devAccIn, nMemoryLocationsIn, queue)),
-          dPhiChangeMaxs_buf(allocBufWrapper<FPX>(devAccIn, nMemoryLocationsIn, queue)),
-          innerLowerModuleIndices_buf(allocBufWrapper<uint16_t>(devAccIn, nMemoryLocationsIn, queue)),
-          outerLowerModuleIndices_buf(allocBufWrapper<uint16_t>(devAccIn, nMemoryLocationsIn, queue)),
-          seedIdx_buf(allocBufWrapper<unsigned int>(devAccIn, maxPixelSegments, queue)),
-          mdIndices_buf(allocBufWrapper<unsigned int>(devAccIn, nMemoryLocationsIn * 2, queue)),
-          nMemoryLocations_buf(allocBufWrapper<unsigned int>(devAccIn, 1, queue)),
-          innerMiniDoubletAnchorHitIndices_buf(allocBufWrapper<unsigned int>(devAccIn, nMemoryLocationsIn, queue)),
-          outerMiniDoubletAnchorHitIndices_buf(allocBufWrapper<unsigned int>(devAccIn, nMemoryLocationsIn, queue)),
-          charge_buf(allocBufWrapper<int>(devAccIn, maxPixelSegments, queue)),
-          superbin_buf(allocBufWrapper<int>(devAccIn, maxPixelSegments, queue)),
-          nSegments_buf(allocBufWrapper<unsigned int>(devAccIn, nLowerModules + 1, queue)),
-          totOccupancySegments_buf(allocBufWrapper<unsigned int>(devAccIn, nLowerModules + 1, queue)),
-          pLSHitsIdxs_buf(allocBufWrapper<uint4>(devAccIn, maxPixelSegments, queue)),
-          pixelType_buf(allocBufWrapper<PixelType>(devAccIn, maxPixelSegments, queue)),
-          isQuad_buf(allocBufWrapper<char>(devAccIn, maxPixelSegments, queue)),
-          isDup_buf(allocBufWrapper<char>(devAccIn, maxPixelSegments, queue)),
-          partOfPT5_buf(allocBufWrapper<bool>(devAccIn, maxPixelSegments, queue)),
-          ptIn_buf(allocBufWrapper<float>(devAccIn, maxPixelSegments, queue)),
-          ptErr_buf(allocBufWrapper<float>(devAccIn, maxPixelSegments, queue)),
-          px_buf(allocBufWrapper<float>(devAccIn, maxPixelSegments, queue)),
-          py_buf(allocBufWrapper<float>(devAccIn, maxPixelSegments, queue)),
-          pz_buf(allocBufWrapper<float>(devAccIn, maxPixelSegments, queue)),
-          etaErr_buf(allocBufWrapper<float>(devAccIn, maxPixelSegments, queue)),
-          eta_buf(allocBufWrapper<float>(devAccIn, maxPixelSegments, queue)),
-          phi_buf(allocBufWrapper<float>(devAccIn, maxPixelSegments, queue)),
-          score_buf(allocBufWrapper<float>(devAccIn, maxPixelSegments, queue)),
-          circleCenterX_buf(allocBufWrapper<float>(devAccIn, maxPixelSegments, queue)),
-          circleCenterY_buf(allocBufWrapper<float>(devAccIn, maxPixelSegments, queue)),
-          circleRadius_buf(allocBufWrapper<float>(devAccIn, maxPixelSegments, queue)) {
-      alpaka::memset(queue, nSegments_buf, 0u);
-      alpaka::memset(queue, totOccupancySegments_buf, 0u);
-      alpaka::memset(queue, partOfPT5_buf, false);
-      alpaka::memset(queue, pLSHitsIdxs_buf, 0u);
-    }
+  GENERATE_SOA_LAYOUT(SegmentsMemSoALayout,
```

Comment:
I'd not restrict the names/modifiers to just 3 characters Mem and Mod.
I'm not sure `Mem` is needed, here each entry is (potentially) a segment, so `SegmentsSoALayout`. For the Mod, each entry is a module detail, perhaps call it `Occupancy` (`SegmentsOccupancySoALayout`)?
`Pix` could be `Pixel`, perhaps even `PixelSegmentsSoALayout`

Reply:
Yeah, that sounds much better